### PR TITLE
use g_auto helpers for automatic cleanup 

### PR DESCRIPTION
**Pull request by pH5**
Bump the required GLib version to 2.45.8 and use g_auto helpers for automatic cleanup.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -768,8 +756,6 @@ static gchar* info_formatter_json_base(RaucManifest *manifest, gboolean pretty)
 	str = json_generator_to_data(gen, NULL);
 
 	json_node_free(root);
```
Comment (mnhu) [inline src/main.c]: Without looking into the json-api: Should the root pointer still be freed when declared as autoptr?

File: src/main.c [hunk 2]
```diff
@@ -1193,10 +1174,6 @@ static gchar* r_status_formatter_json(gboolean pretty)
 	json_generator_set_pretty(gen, pretty);
 	str = json_generator_to_data(gen, NULL);
 
-	json_node_free(root);
```
Comment (mnhu) [inline src/main.c]: Here the json root node pointer is freed as opposed to my previous comment.

File: src/main.c [hunk 3]
```diff
@@ -1654,7 +1628,7 @@ static void cmdline_handler(int argc, char **argv)
 	g_free(text);
 
 done:
-	g_clear_pointer(&context, g_option_context_free);;
+	return;
```
Comment (mnhu) [inline src/main.c]: Why the return statement here at the very end of a void function?

File: src/main.c (hunk 1 above)
Comment (pH5) [reply src/main.c]: Absolutely not, thanks for catching this!
json_node_free is the autoptr cleanup function registered for the JsonNode type, so I have to remove this as well.

File: src/main.c (hunk 3 above)
Comment (pH5) [reply src/main.c]: So I can keep the "done:" label and not have to change all the "goto done;"s in this function in this patch.

**Comment by ejoerns** [comment]
Just added two further fixes that I found during second review

File: src/manifest.c [hunk 4]
```diff
@@ -136,12 +136,9 @@ static gboolean parse_image(GKeyFile *key_file, const gchar *group, RaucImage **
 	g_key_file_remove_group(key_file, group, NULL);
 
 	res = TRUE;
-	*image = iimage;
+	*image = g_steal_pointer(&iimage);
 
 out:
```
Comment (michaelolbrich) [inline src/manifest.c]: the label is no longer necessary. Just return above. And remove `res`.

File: src/manifest.c [hunk 5]
```diff
@@ -299,20 +296,15 @@ static gboolean parse_manifest(GKeyFile *key_file, RaucManifest **manifest, GErr
 	g_strfreev(groups);
 
 	res = TRUE;
+	*manifest = g_steal_pointer(&raucm);
 free:
```
Comment (michaelolbrich) [inline src/manifest.c]: same here.

File: src/manifest.c [hunk 6]
```diff
@@ -338,14 +330,13 @@ gboolean load_manifest_mem(GBytes *mem, RaucManifest **manifest, GError **error)
 	}
 
 out:
```
Comment (michaelolbrich) [inline src/manifest.c]: and here.

File: src/manifest.c [hunk 7]
```diff
@@ -473,17 +460,13 @@ gboolean save_manifest_file(const gchar *filename, RaucManifest *mf, GError **er
 
 		if (file->filename)
 			g_key_file_set_string(key_file, group, "filename", file->filename);
-
-		g_free(group);
 	}
 
 	res = g_key_file_save_to_file(key_file, filename, NULL);
```
Comment (michaelolbrich) [inline src/manifest.c]: just return here.

**Comment by michaelolbrich** [comment]
There are several more res / goto out combinations that are no longer needed.

**Comment by ejoerns** [comment]
@michaelolbrich Yes, I've seen them but decided to remove them step by step in separate patches to not block this more than required and to not let the patches grow too much.

But maybe we can add fixes for some very obvious ones with low loc impact.

**Comment by ejoerns** [comment]
@michaelolbrich, @pH5 I've pushed a fixup now. Please let me know if your are ok with this and if we can merge.

**Comment by michaelolbrich** [comment]
The fixup is ok.
I still feel that the now unnecessary gotos should be removed as well, but nothing looks broken.

**Comment by ejoerns** [comment]
@michaelolbrich I've already prepared some patches for that, but not entirely finished. If you promise to have a look on these too when I finish and push them, we can still do it in this PR :)

**Comment by michaelolbrich** [comment]
@ejoerns I think we can merge this PR. Just ping me when the followup PR is ready

**Comment by ejoerns** [comment]
Ok, sounds like a plan! Then approve it, to give me a good feeling when pushing the big button ;)

**Comment by ejoerns** [comment]
Squashed and rebased onto master

**Pull request merged**